Gir.toml

```diff
@@ -605,6 +605,10 @@ status = "generate"
     [[object.signal]]
     name = "writable-change-event"
     inhibit = true
+    [[object.function]]
+    name="set_*"
+        [object.function.return]
+        bool_return_is_error = "Can't set readonly key"
```

Comment:
This doesn't return `false` if the key was of the wrong type? What happens if you try to set a boolean on an integer key?

Reply:
I'm pretty sure GLib just SIGTRAPs then

Reply:
And the docs say FALSE == not writeable, so I interpreted that as readonly: https://developer.gnome.org/gio/stable/GSettings.html#g-settings-set-int

Reply:
Yup,
```
(process:13224): GLib-GIO-ERROR **: 23:42:07.005: Settings schema 'com.github.gtk-rs.test' does not contain a key named 'no-such-key'
test settings::test::bool_set_get ... ok
error: process didn't exit successfully: `/home/rasmus/Projects/gio/target/debug/deps/gio-8e55b7fc5fe6fb99` (signal: 5, SIGTRAP: trace/breakpoint trap)
```

Reply:
Alright, at this point I'm a bit unsure why we even return Option<Variant> from SettingsExt::get_value()

Reply:
`get_value()` simply explodes too if the key does not exist? I.e. it can't possibly return `None`? In that case it should indeed not return an `Option` and this has to be fixed here too, leaving only the type mismatch as a valid error.

If you call `get_string()` on a bool setting, it also explodes? In that case all these functions should simply return their value and all the error handling added is not actually useful because of how the GLib API works. You could then simply panic in the getter if the type is wrong as that's consistent with how the GLib API works.

Sorry for the confusion but I wasn't aware of this behaviour of the GLib API, that it simply kills the process if the type is wrong or the key does not exist. Not very user-friendly API.

Reply:
> If you call get_string() on a bool setting, it also explodes? In that case all these functions should simply return their value and all the error handling added is not actually useful because of how the GLib API works. You could then simply panic in the getter if the type is wrong as that's consistent with how the GLib API works.

Hm, I don't see a problem with keeping the type mismatch error since we can still handle that (since we just get the Value and convert it ourselves). The default GLib expectation _is_ that it's a programmer error if you don't know what kind of key you're getting though, so we can go with that too if you want.
 
> Sorry for the confusion but I wasn't aware of this behaviour of the GLib API, that it simply kills the process if the type is wrong or the key does not exist.

No worries :)

> Not very user-friendly API.

Well, usually it should be easy enough to guarentee that your GSchema is alright, the problem is when one wants to fetch some settings from foreign schemas (e.g. from the schema contained in gsettings-desktop-schemas, which are _always_ a pain to upgrade for distros because you have to bump the whole desktop at once or everything just SIGTRAPs because some key has been changed :/). As such I feel like we should keep at least the type error, the user can always just `unwrap()` if he wants to. I suppose we could even implement the `No such key` error by checking the key against `SettingsExt::list_keys()`, but that is potentially too expensive and I don't think list_keys is really meant for that purpose.

Reply:
> Hm, I don't see a problem with keeping the type mismatch error since we can still handle that (since we just get the Value and convert it ourselves). The default GLib expectation is that it's a programmer error if you don't know what kind of key you're getting though, so we can go with that too if you want.

In all other functions a missing key or type mismatch causes the process to be killed be GLib. So we should try to be consistent with that and at least panic :)

> I suppose we could even implement the No such key error by checking the key against SettingsExt::list_keys(), but that is potentially too expensive and I don't think list_keys is really meant for that purpose.

Yes, that's something an application could do. We shouldn't diverge that much from the C API.

Reply:
Alright, sure 👍

Reply:
Panicking here now